test/e2e/tests/test_endpoint.py

```diff
@@ -51,6 +52,7 @@ def single_container_model(name_suffix):
         spec_file="xgboost_model",
         replacements=replacements,
     )
+    model_resource = k8s.wait_resource_consumed_by_controller(model_reference)
```

Comment:
same comment as above... you are already calling the equivalent of `k8s.wait_resource_consumed_by_controllers` below for all resources involved in these tests...

Reply:
@ryansteakley wanted to follow up on this comment. Please note that when I said you are already calling the equivalent of `k8s.wait_resource_consumed_by_controller`, I was referring to the fact that in test_endpoint.py, you are calling `k8s.wait_on_condition`:

https://github.com/aws-controllers-k8s/sagemaker-controller/blob/17c03a5e60d360240ede43928f645616325bc0b4/test/e2e/tests/test_endpoint.py#L232

after creating the CR in kubernetes. And that is essentially the equivalent of `k8s.wait_resource_consumed_by_controller`